### DISCUSSION TITLE
Cant fetch messages in sub groups of main discord

### DISCUSSION THREAD

**Discussion by webupnz** [Tyrrrz/DiscordChatExporter#972] [General]
Love this tool as its very handy for my note taking in studying.
One issue I seem to have is certain sections of a discord that I have access to wont work with discord chat exporter.
I'm using the filter 'from:Username#1234'
I get a message saying 'No messages found for the specified period'

I've checked the section and there are posts from the person in there but the exporter cant get them.

**Comment by 96-LB** [comment]
Does the username contain spaces or special characters?

**Reply by webupnz** [reply]
yes it has a space ie Person R but I tried with space then no space

**Reply by 96-LB** [reply]
Put quotes around the name, i.e. `from:"Person R#1234"` or `from:'Person R#1234'`

**Comment by webupnz** [comment]
Thanks, while that works for the gen chat section it doesnt work on some of the other sections.
On the actual discord i can see the posts in the sub section and if i run the exact same search that works on gen chat it fails on the others

**Reply by 96-LB** [reply]
When you say subsection, are you referring to threads? See below for example:
![image](https://user-images.githubusercontent.com/21089343/209014178-00a936fa-92b3-4399-8a7d-ce46ddf22fb6.png)

If so, then DiscordChatExporter doesn't export content of threads when you export the parent channel. You have to export the threads separately, which can only be done in the CLI version.

**Reply by webupnz** [reply]
ah yes its a thread under a section.
How do i install the CLI version?

**Reply by 96-LB** [reply]
Pick one of the downloads in the CLI row [here](https://github.com/Tyrrrz/DiscordChatExporter#download)

**Reply by webupnz** [reply]
i downloaded the stable release CLI but no idea how to run it.

**Reply by 96-LB** [reply]
The Wiki has instructions on how to get started - you should check those out. To export a thread, you can copy its ID the same way you would for any other channel, and follow the directions for exporting a channel.

**Reply by webupnz** [reply]
Opening DiscordChatExporter.Cli.exe opens Visual Studio

**Reply by webupnz** [reply]
looks like i have it setup. How do i add the filter to only one username?

**Reply by 96-LB** [reply]
Use the `--filter` option, as specified in the Wiki

**Reply by webupnz** [reply]
wow thanks, my brain nearly exploded doing all the setup but it seems i have it working!
